app/api/srch/search.rb

```diff
@@ -115,6 +115,30 @@ class Search < Grape::API
         sresult
       end
 
+      # Request URL should be /api/srch/locations?srchString=QRY[&seq=KEYCOUNT&showCount=NUM_ROWS&pageNum=PAGE_NUM]
+      # Note: Query(QRY as above) must have latitude and longitude as srchString=lat,lon
+      desc 'Perform a search of documents having nearby latitude and longitude tag values',
+                                                                    hidden: false,
+                                                                           is_array: false,
+                                                                           nickname: 'srchGetLocations'
+      params do
+        requires :srchString, type: String, documentation: { example: 'Spec' }
+        optional :seq, type: Integer, documentation: { example: 995 }
+        optional :showCount, type: Integer, documentation: { example: 3 }
+        optional :pageNum, type: Integer, documentation: { example: 0 }
+      end
+      get :locations do
+        sresult = DocList.new
+        unless params[:srchString].nil? || params[:srchString] == 0 || !(params[:srchString].include? ",")
+          sservice = SearchService.new
```

Comment:
params[:srchString].nil? || params[:srchString] == 0
this particular code is kind of used in multiple places in this file. can't be move it to a private method and re-use it relevant place ?

Reply:
I think so -- but perhaps independently from this PR so we address code repetition as its own issues. Would you mind opening a new issue to address this? I agree that the services and api code in particular can be enormously compacted without loss of organization.  Thank you!